src/pybind/mgr/dashboard/services/nvmeof_client.py

```diff
@@ -28,7 +28,18 @@ def __init__(self):
             self.gateway_addr = list(
                 NvmeofGatewaysConfig.get_gateways_config()["gateways"].values()
             )[0]["service_url"]
-            self.channel = grpc.insecure_channel("{}".format(self.gateway_addr))
+            server_cert = NvmeofGatewaysConfig.get_server_cert()
+            client_key = NvmeofGatewaysConfig.get_client_key()
+            client_cert = NvmeofGatewaysConfig.get_client_cert()
```

Comment:
This means that for every client connection (and every API requests involves a fresh client connection), 3 reads will hit the mon KV. Not a bit thing, but given those will rarely change, I'd put them outside the `__init__` method. Any other ideas?

Reply:
for now I moved it outside `__init__` and invoked it as a class-level attribute so that it only invokes when the class is first initialized. Probably I should finish the [other PR](https://github.com/ceph/ceph/pull/55951) to improve this further.

Reply:
Thanks!

Reply:
In future, looks like this is not constant right? Because we'll have to get the cert for each services if there are more than one so making it constant might not work as it seems. So I bought back the previous implementation.